Create Extension: Add dynamic config and Woo version number

psealock opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR wraps the `module.exports` contents in a self invoked async function so that we can `await` any API calls with relevant data. In this case, we gather the latest version of WooCommerce from the wordpress.org API so that wp-env can be correctly configured with the latest version.

Closes https://github.com/woocommerce/platform-private/issues/130

<!-- The next section is mandatory. If your PR doesn't require testing, please indicate that you are purposefully omitting instructions. -->

- [x] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Otherwise, please include detailed instructions on how these changes can be tested (including pre-conditions, configuration, steps to take and expected results). It may help to write your instructions using pseudocode -- as if you're telling a computer how to execute the test. -->

Testing this PR requires a development version of Gutenberg.

1. Pull down Gutenberg and checkout `add/create-block-dynamic-config` from this PR https://github.com/WordPress/gutenberg/pull/46456
2. `npm install` and make sure you are on Node v14 for this step.
3. In this repo, be sure to `pnpm install` as I've added dependencies.
4. Run this command from a new folder.
```
npx ../path/to/gutenberg/packages/create-block -t ../path/to/woocommerce/packages/js/create-extension my-extension-name
```
5. Check the resulting project, it should have a `.wp-env.json` file correctly configured with a download link to the latest WooCommerce.

<!-- End testing instructions -->

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

vedanshujain committed:
Revert "Revert "[COT] Add the orders cache" (#34992)"

This reverts commit 0b5d7ab9ad4ef09fc0ef47c5deddfe2aa4f7e5c3.

psealock committed:
Get latest Woo version to download

roykho reviewed (dismissed):
Tested and working as advertised. Good work! Just need to fix a conflict with `pnpm-lock.yaml` but other than that all good. Approving for now.

vedanshujain committed:
Move the namespace algorithm functions to dedicated CacheNameSpaceTrait.

We intend to use the same namespacing algorithm for HPOS orders cache like we have been using before for posts. Moving this structure to a common trait allow us to use the same logic for both posts and HPOS.

vedanshujain committed:
Allow grouping support in cache engines.

This will allow us to add namespacing support directly in the cache engine.

vedanshujain committed:
Remove namespacing from object cache since its supported by cache engine directly.

vedanshujain committed:
Use HPOS feature flag for order cache as well.

vedanshujain committed:
Apply renaming of WPCacheEngine class.

vedanshujain committed:
Fix unit tests for object cache to accomodate recent changes.

vedanshujain committed:
Applied coding standards.

vedanshujain committed:
Applied coding standards.

vedanshujain committed:
Merge fixes.

vedanshujain committed:
Coding standard fix.

vedanshujain committed:
Remove infinite loop by unlinking cache and datastore.

vedanshujain committed:
Clear cache after deleting post.

vedanshujain committed:
Temporary disable cache before backfilling.

vedanshujain committed:
Coding standard fixes.

vedanshujain committed:
Remove get_from_datastore as it was not being used.

vedanshujain committed:
Use correct class name.

lsinger committed:
Merge branch 'trunk' into add/create-extensions-dynamic-config

lsinger commented:
Resolved conflicts and rebased. 

Using `npx ../path/to/gutenberg/packages/create-block/index.js -t ../path/to/woocommerce/packages/js/create-extension my-extension-name` to run it this sort of works for me. The project being generated does have issues, though -- the following line throws a fatal error: 

`require_once plugin_dir_path( __FILE__ ) . '/vendor/autoload_packages.php';`

This is probably an issue with https://github.com/WordPress/gutenberg/pull/46456 I suppose? 🤔

psealock commented:
>the following line throws a fatal error:

>require_once plugin_dir_path( __FILE__ ) . '/vendor/autoload_packages.php';

Sounds like `compose install` didn't run. It should be a postinstall script.
